qa/rpc-tests/excessive.py

```diff
@@ -12,17 +12,231 @@
+
+        if self.extended:  # this test checks the behavior of > 1MB blocks with excessive transactions.  it takes a LONG time to generate and propagate 1MB+ txs.
+  	  # Create a LOT of UTXOs for the next test
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+          logging.info("Create lots of UTXOs...")
+          n=0
+
+          for w in wallet:
+            split_transaction(self.nodes[0], [w], addrs[n:100+n])
+            n+=100
+            if n >= len(addrs): n=0
+	
+          self.nodes[0].generate(1)
+	  self.sync_all()
+
+          logging.info("Building > 1MB block...")
+          self.nodes[0].set("net.excessiveTxn=1000000")  # Set the excessive transaction size larger for this node so we can generate an "excessive" block for the other nodes
+        
+          wallet = self.nodes[0].listunspent()
+          wallet.sort(key=lambda x: x["amount"],reverse=False)
+
+          # Generate 1 MB worth of transactions        
+          size = 0
+          count = 0
+          while size < 1000000:
+            count+=1
+            utxo = wallet.pop()
+            outp = {}
+	    outp[addrs[count%len(addrs)]] = utxo["amount"]
+            txn = self.nodes[0].createrawtransaction([utxo], outp)
+            signedtxn = self.nodes[0].signrawtransaction(txn)
+            size += len(binascii.unhexlify(signedtxn["hex"]))
+            self.nodes[0].sendrawtransaction(signedtxn["hex"])
+
+          # Now generate a > 100kb transaction & mine it into a > 1MB block
+
+          self.nodes[0].setminingmaxblock(2000000)
+          wallet.sort(key=lambda x: x["amount"],reverse=True)
+	  (tx, vin, vout, txid) = split_transaction(self.nodes[0],wallet[0:2000],[addrs[0]],txfeePer=60)
+          logging.debug("Transaction Length is: ", len(binascii.unhexlify(tx)))
+          # assert(binascii.unhexlify(tx) > 100000) # txn has to be big for the test to work
+	  origCounts = [ x.getblockcount() for x in self.nodes ]
+          mpool = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          print mpool
+          largeBlock = self.nodes[0].generate(1)
+          time.sleep(10) # can't sync b/c nodes won't be in sync
+          mpool = [ (lambda y: (y["size"],y["bytes"]))(x.getmempoolinfo()) for x in self.nodes]
+          print mpool
+          counts = [ x.getblockcount() for x in self.nodes ]
+          latest = counts[0]
+          excess = latest-1
+          #assert_equal(counts, [latest,excess,excess,excess]) 
+
+          logging.info("Syncing node1")
+          largeBlock2 = self.nodes[0].generate(1)
+          while 1:
+            counts = [ x.getblockcount() for x in self.nodes ]
+            if counts[0] == counts[1]:  # this is a large block with lots of tx so can take a LONG time to sync on one computer
+              break
+            time.sleep(10) # can't sync b/c nodes won't be in sync
```

Comment:
see my comment on fixed sleeps above
the while: logic used here is repeated twice with almost the same loop contents (only node numbers used for comparisons differ).
Consider factoring this sync loop out into a method.